### PR TITLE
OOP school library: unit tests (PR)

### PR DESCRIPTION
**Pull request by Doheera-kosi**
# Milestone Learning objectives:
- [ ] Write unit tests using Rspec.
- [ ] Understand an existing codebase in a way that allows for adding tests.

## General requirements
- No linter errors
- Used correct Gitflow
- Documented work in a professional way

## Ruby requirements
- Follow our list of best practices for Ruby

## Project requirements
- [ ] Add unit tests for all the methods in all the classes:
- Book
- Person
- Student
- Teacher
- Classroom
- Rental

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      expect(book.title).to eql('Evan')
+    end
+    it 'output book author' do
+      expect(book.author).to eql('Sajeel')
+    end
+    it 'is an instance of Book' do
+      expect(book).to be_an_instance_of(Book)
+    end
+  end
+end
```
Comment (Felix45) [inline spec/book_spec.rb]: - [ Required ] Please consider adding a `unit test` for the `add_rentals` method in class `Book`, the project requirements state that you create unit tests for all the methods implemented in each class, currently there is no unit test for the `add_rentals` method. You can do this by following the example below
```Ruby
require_relative '../classes/book'
require_relative '../classes/rentals'
require_relative '../classes/student'

describe Book do
    book = Book.new('Think Big', 'Ben Carson')
    it 'adding one rental to book should have length of one' do
      book.add_rentals(Student.new(20, true, 'Felix Ouma'), '2022/08/30',)
      expect(book.rentals.length).to eq(1)
    end
end
```

Screenshot from 2023-01-11 23-48-51

File: spec/classroom_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require_relative '../classes/classroom'
+
+describe Classroom do
+  classroom = Classroom.new('Science')
+  describe 'Classroom Information' do
+    it 'should have a label of Science' do
+      expect(classroom.label).to eql('Science')
+    end
+  end
+end
```
Comment (Felix45) [inline spec/classroom_spec.rb]: - [ Required ] Please consider adding a `unit test` for the `add_student` method in class `Classroom`, the project requirements state that you create unit tests for all the methods implemented in each class, currently there is no unit test for the `add_student` method. You can do this by following the example below
```Ruby 
require_relative '../classes/classroom'
require_relative '../classes/student'

describe Classroom do
  classroom = Classroom.new('Science')
  describe 'Classroom Information' do
    it 'student should be added to class' do
      classroom.add_student(Student.new(14, false, 'John'))
      expect(classroom.students.length).to eq(1)
    end
  end
end
```

Screenshot from 2023-01-11 23-48-51

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      expect(person.correct_name).to eql('Sajeel')
+    end
+    it 'should have an age of 28' do
+      expect(person.age).to eql(28)
+    end
+    it 'should be able to use services' do
+      expect(person.can_use_services?).to eql(true)
+    end
+  end
+end
```
Comment (Felix45) [inline spec/person_spec.rb]: - [ Required ] Please consider adding a `unit test` for the `add_rentals` method in class `Person`, the project requirements state that you create unit tests for all the methods implemented in each class, currently there is no unit test for the `add_rentals` method in class `Person`. 

Screenshot from 2023-01-11 23-48-51

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @Doheera-kosi  & @Sajeelzafar , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :heavy_check_mark: 
- [ ] :clap: Descriptive pull request title and summary :heavy_check_mark: 
- [ ] :clap: All linter checks are set up correctly and passing with green check marks :heavy_check_mark: 
- [ ] :clap: Good descriptive commit messages :heavy_check_mark: 
- [ ] :clap: Git flow is correctly used :heavy_check_mark: 
- [ ] :clap:  Professional readme file :heavy_check_mark: 



## Required Changes ♻️
- [x] Please consider adding unit tests for all the methods created when implementing the decorator design pattern since this is a project requirement. Please see the screenshot below for more details on this.

Screenshot from 2023-01-11 23-48-51

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @Doheera-kosi and @Sajeelzafar , 

To highlight 👍🏽 
✔️ You added more tests
✔️ No linter errors

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**